src/main/java/mil/dds/anet/resources/SavedSearchResource.java

```diff
@@ -34,14 +35,33 @@ public SavedSearch createSavedSearch(@GraphQLRootContext GraphQLContext context,
       @GraphQLArgument(name = "savedSearch") SavedSearch savedSearch) {
     Person user = DaoUtils.getUserFromContext(context);
     savedSearch.setOwnerUuid(user.getUuid());
+    Double maxPriority = dao.getMaxPriorityForOwner(user.getUuid());
+    savedSearch.setPriority(maxPriority == null ? 0.0 : maxPriority + 1.0);
+    if (savedSearch.getDisplayInHomepage()) {
+      Double maxHomepagePriority = dao.getMaxHomepagePriorityForOwner(user.getUuid());
+      savedSearch
+          .setHomepagePriority(maxHomepagePriority == null ? 0.0 : maxHomepagePriority + 1.0);
+    }
     final SavedSearch created = dao.insert(savedSearch);
     AnetAuditLogger.log("SavedSearch {} created by {}", created, user);
     return created;
   }
 
   @GraphQLQuery(name = "mySearches")
   public List<SavedSearch> getMySearches(@GraphQLRootContext GraphQLContext context) {
-    return dao.getSearchesByOwner(DaoUtils.getUserFromContext(context));
+    return dao.getSearchesByOwner(DaoUtils.getUserFromContext(context)).stream()
+        .sorted(
+            Comparator.comparing(SavedSearch::getPriority, Comparator.nullsLast(Double::compareTo)))
+        .toList();
+  }
+
+  @GraphQLQuery(name = "myHomepageSearches")
+  public List<SavedSearch> getMyHomepageSearches(@GraphQLRootContext GraphQLContext context) {
+    Person user = DaoUtils.getUserFromContext(context);
+    return dao.getSearchesByOwner(user).stream()
+        .filter(s -> Boolean.TRUE.equals(s.getDisplayInHomepage())).sorted(Comparator
+            .comparing(SavedSearch::getHomepagePriority, Comparator.nullsLast(Double::compareTo)))
+        .toList();
```

Comment:
These two can be combined into a single method by introducing a parameter `boolean forHomepage` (and also adding that to `dao.getSearchesByOwner`).